src/contracts/GPv2Settlement.sol

```diff
@@ -27,6 +35,16 @@ contract GPv2Settlement {
     /// @param uniswapFactory_ The Uniswap factory to act as the AMM for this
     /// GPv2 settlement contract.
     constructor(IUniswapV2Factory uniswapFactory_) public {
+        // NOTE: Currently, the only way to get the chain ID in solidity is
+        // using assembly.
+        uint256 chainId;
+        assembly {
+            chainId := chainid()
+        }
+
+        replayProtection = keccak256(
+            abi.encodePacked(DOMAIN_SEPARATOR, uint64(chainId), address(this))
```

Comment:
Is the domain separator really needed if we include the contract address?

Reply:
Maybe its nice to add some salt?

Reply:
The only other reason I can think for including a domain separator would be if there were another contract `Contract is GPv2Settlement`. The domain separator is a polite way to create some signature replay protection data in such a way that does not collide with potential replay protection data for the smart contract inheriting `GPv2Settlement`.

Again, there are 10000000 other ways around this, but it does make the replay protection signature mixer a bit more "unique".

Reply:
I'd keep the salt in: it only adds a small overhead during deployment, but it makes it less likely to overlap with other services that may use a format of the form `hash(uint64, address)` for something else that isn't a domain separator. (This is also why it's nice to have unpacked bits, so that you don't have to think too much about how other people would use your same bitstring but split it in another way in some other context.)

Reply:
> This is also why it's nice to have unpacked bits

Do you mean using `abi.encode` vs `abi.encodePacked`?

Reply:
I modified the code to use `abi.encode` instead of `abi.encodePacked`.